cmd/server/cadence/cadence.go

```diff
@@ -135,13 +139,86 @@ func BuildCLI() *cli.App {
 		{
 			Name:    "start",
 			Aliases: []string{""},
-			Usage:   "start cadence notification service",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  "services",
+					Value: "receiver, notifier",
+					Usage: "set up webhook test endpoint",
+				},
+			},
+			Usage: "start cadence notification service",
 			Action: func(c *cli.Context) {
-				startHandler(c)
+				var wg sync.WaitGroup
+				services := getServices(c)
+
+				for _, service := range services {
+					wg.Add(1)
+					go launchService(service, c)
+				}
+
+				wg.Wait()
 			},
 		},
 	}
-
 	return app
+}
+
+func launchService(service string, c *cli.Context) {
+	switch service {
+	case "notifier":
+		startHandler(c)
+		break
+	case "receiver":
+		startTestWebhookEndpoint()
+		break
+	default:
+		log.Printf("Invalid service: %v", service)
+	}
+}
+
+func getServices(c *cli.Context) []string {
+	val := strings.TrimSpace(c.String("services"))
+	tokens := strings.Split(val, ",")
+
+	if len(tokens) == 0 {
+		return []string{"notifier"}
+	}
+
+	services := []string{}
+	for _, token := range tokens {
+		t := strings.TrimSpace(token)
+		services = append(services, t)
+	}
 
+	return services
+}
+
+func startTestWebhookEndpoint() {
+	http.HandleFunc("/", logIncomingRequest)
+
+	fmt.Printf("Starting server for testing...\n")
+	if err := http.ListenAndServe(":8080", nil); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func logIncomingRequest(w http.ResponseWriter, r *http.Request) {
+	if r.URL.Path != "/" {
+		http.Error(w, "404 not found.", http.StatusNotFound)
+		return
+	}
+
+	switch r.Method {
+	case "POST":
+		var body []byte
+		body, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			log.Printf("[Failed to read request body]: %v", err.Error())
+		}
+
+		log.Printf("[Test server incoming request]: %v", string(body))
```

Comment:
Can we also log the url path? So that we can verify the notifier is honoring the path defined in the config

Reply:
add log for url path, but I think if the path is incorrect, it will return 404 earlier in line 206-209

Reply:
I think you should remove line 206-209. Is there any reason to have them?

Reply:
I was thinking we could just enforce a certain url path at first, but fine to remove

Reply:
actually that’s a good idea to enforce it. But to be more generic, can we do “/callback” ? Just to make sure our notifier can invoke any path other than root “/“.